### PR TITLE
[bitnami/etcd] Add ETCD_LISTEN_METRICS_URLS

### PR DESCRIPTION
**Pull request by svenwiltink**
### Description of the change
Allow setting ETCD_LISTEN_METRICS_URLS

### Benefits

Allow the scraping of metrics without tls certificates on a secured deployment

### Applicable issues

- fixes https://github.com/bitnami/charts/issues/18451

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by svenwiltink** [comment]
should the metrics also be added to the (headless) services?

**Comment by svenwiltink** [comment]
Some context: we want to monitor hundreds of etcd deployments without maintaining an equal amount of certificates for prometheus.

File: bitnami/etcd/values.yaml [hunk 1]
```diff
@@ -725,11 +727,17 @@ metrics:
   ## @param metrics.enabled Expose etcd metrics
   ##
   enabled: false
+
+
+  ## @param metrics.useSeparateEndpoint Use a separate endpoint for exposing metrics
+  #
+  useSeparateEndpoint: false
+
```
Comment (FraPazGal) [inline bitnami/etcd/values.yaml:735]: Going by the style of the rest of the file, there is no need for newlines between parameters of a subgroup:

```suggestion
  ## @param metrics.useSeparateEndpoint Use a separate endpoint for exposing metrics
  ##
  useSeparateEndpoint: false
```

File: bitnami/etcd/values.yaml [hunk 2]
```diff
@@ -286,10 +286,12 @@ lifecycleHooks: {}
 ## etcd container ports to open
 ## @param containerPorts.client Client port to expose at container level
 ## @param containerPorts.peer Peer port to expose at container level
+## @param containerPorts.metrics Metrics port to expose at container level when metrics are enabled
```
Comment (FraPazGal) [inline bitnami/etcd/values.yaml:289]: We should note this port will only be used if `metrics.useSeparateEndpoint=true`

File: bitnami/etcd/templates/networkpolicy.yaml [hunk 3]
```diff
@@ -78,7 +78,7 @@ spec:
     {{- if .Values.metrics.enabled }}
     # Allow prometheus scrapes for metrics
     - ports:
-        - port: 2379
+        - port: {{ .Values.metrics.useSeparateEndpoint | ternary .Values.containerPorts.metrics .Values.containerPorts.client}}
```
Comment (FraPazGal) [inline bitnami/etcd/templates/networkpolicy.yaml:81]:
```suggestion
        - port: {{ .Values.metrics.useSeparateEndpoint | ternary .Values.containerPorts.metrics .Values.containerPorts.client }}
```

File: bitnami/etcd/templates/podmonitor.yaml [hunk 4]
```diff
@@ -21,7 +21,7 @@ metadata:
   {{- end }}
 spec:
   podMetricsEndpoints:
-    - port: client
+    - port: {{ .Values.metrics.useSeparateEndpoint | ternary "metrics"  "client" }}
```
Comment (FraPazGal) [inline bitnami/etcd/templates/podmonitor.yaml:24]:
```suggestion
    - port: {{ .Values.metrics.useSeparateEndpoint | ternary "metrics" "client" }}
```

**Review by FraPazGal — COMMENTED** [review]
Hi @svenwiltink, thanks for another PR! I've added a few comments as a review, most of them just style-related.

Besides that, we do need to include the new por to the services templates as you said. Otherwise, the por wont be accesible via the svc's IP not internal headless endpoint.

**Comment by svenwiltink** [comment]
Thanks for the quick review! I've fixed the styling and added the ports to the services when the separate endpoint is exposed

**Pull request merged**